Pipes - Kate Evans-Spitzer - Solar System.rb

```diff
@@ -0,0 +1,240 @@
+#########################
+###### DEFINITIONS ######
+#########################
+
+
+class SolarSystem
+
+  attr_reader :system_name, :num_planets, :planets, :planet_list
+
+  def initialize(name, planets) 
+    @system_name = name.to_s
+    @planets = {}
+    planets.each {|new_planet| @planets[new_planet.name] = new_planet}
+    @num_planets = @planets.length
+    # puts "@planets['Mars'] looks like: #{@planets["Mars"]}"            # => #<Planet:0x007fcefb89cd90>
+    # puts "@planets['Mars'].name looks like: #{@planets["Mars"].name}"  # => Mars
+  end
+
+  def add_planet(*planet)         # Adds one or more Planet objects to solar system
+    planet.each {|new_planet| @planets[new_planet.name] = new_planet}
+    @num_planets = @planets.length
```

Comment:
It's interesting to me that you stored the number of planets in a variable. As we can see, you have to remember to update the value every time the set of planets changes. This isn't necessarily a bad thing, but it's an extra piece for you the developer to remember. For example, it would be easy to write a `remove_planet` method and forget to update the count.

A cleaner way to do it might be to build an instance method that provides direct access to `@planets.length`, something like:

```ruby
def num_planets
  return @planets.length
end
```

This has the same effect, without requiring you to keep track of an extra thing.